src/components/DeleteModalContainer.tsx

```diff
@@ -1,21 +1,22 @@
 import { RefObject } from 'react';
 import { Task } from '../utils/types';
 import DeleteModalUI from './DeleteModalUI';
+import { useTaskManager } from '../hooks/useTaskManager';
 
 interface DeleteModalContainerProps {
   deleteModalRef: RefObject<HTMLDialogElement>;
   tasks: Task[];
-  deleteAll: () => Promise<void>;
 }
 
 function DeleteModalContainer({
   deleteModalRef,
   tasks,
-  deleteAll,
 }: DeleteModalContainerProps) {
+  const { deleteAllTasksMutation } = useTaskManager();
+
   //Handle deleting all tasks
-  const handleTasksDelete = async () => {
-    await deleteAll();
+  const handleTasksDelete = () => {
+    deleteAllTasksMutation.mutate();
```

Comment:
🪡 Nitpick: maybe use async

Reply:
Async wouldn't work here because the `mutate()` function doesn't return anything. I could use `mutateAsync()` but that would only make sense if I had multiple mutations I wanted to trigger and they each needed to be awaited. (based on this article: [mutate vs mutateAsync](https://tkdodo.eu/blog/mastering-mutations-in-react-query#:~:text=Mutate%20or%20MutateAsync&text=mutate%20doesn't%20return%20anything,the%20result%20of%20the%20mutation.) )